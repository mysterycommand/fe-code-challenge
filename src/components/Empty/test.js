import React from 'react';
import Empty from './';

describe('Empty', () => {
  it('should be empty', () => {
    expect(<Empty />).toMatchSnapshot();
  });
});
