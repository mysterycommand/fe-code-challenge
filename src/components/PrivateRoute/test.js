import React from 'react';

import PrivateRoute from './';
import Empty from '../Empty';

describe('PrivateRoute', () => {
  it('should do the right thing based on props', () => {
    [
      {
        component: Empty,
      },
      {
        component: Empty,
        isAuthenticated: true,
      },
      {
        component: Empty,
        isAuthenticated: true,
        redirectTo: '/glerb',
      },
      {
        component: Empty,
        redirectTo: '/glerb',
      },
    ].forEach(props => {
      expect(<PrivateRoute {...props} />).toMatchSnapshot();
    });
  });
});
