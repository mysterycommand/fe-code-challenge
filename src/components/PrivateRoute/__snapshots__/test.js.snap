// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrivateRoute should do the right thing based on props 1`] = `
<PrivateRoute
  component={[Function]}
  isAuthenticated={false}
  redirectTo="/"
/>
`;

exports[`PrivateRoute should do the right thing based on props 2`] = `
<PrivateRoute
  component={[Function]}
  isAuthenticated={true}
  redirectTo="/"
/>
`;

exports[`PrivateRoute should do the right thing based on props 3`] = `
<PrivateRoute
  component={[Function]}
  isAuthenticated={true}
  redirectTo="/glerb"
/>
`;

exports[`PrivateRoute should do the right thing based on props 4`] = `
<PrivateRoute
  component={[Function]}
  isAuthenticated={false}
  redirectTo="/glerb"
/>
`;
